juju/juju
# fix(offer): improve error message when updating offers with removed endpoints

Juju offer is an upsert operation but this might misleading to users. Since we are not changing this behavior right now, this patch at least improves the error message when *updating* an offer and in this update an endpoint (with active consumers) is removed.
The new error message should give the user the hint that they are actually updating the offer and therefore if they don't want to remove the previously added endpoints, they should be added as a list in the passed arguments. This behavior is already documented.


## Checklist


## QA steps

Bootstrap and add two models; one with influxdb on it and other with grafana:

```
juju bootstrap localhost c
juju add-model m1
juju deploy grafana
juju add-model m2
juju deploy influxdb
```
Offer and integrate grafana and influxdb:
```
juju offer influxdb:grafana-source
juju switch m1
juju integrate grafana admin/m2.influxdb
```
Now, try to update the offer and the correct error should be displayed:
```
juju switch m2
juju offer influxdb:query
ERROR cannot update application offer "influxdb": updating offer influxdb:query would remove endpoint "grafana-source" which has active consumers
```


## Links


**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2048870

**Jira card:** JUJU-6124
